### PR TITLE
Update jsf-2.3 to use MyFaces 2.3.10

### LINKED ISSUE #21501: Update the jsf-2.3 feature to MyFaces 2.3.10
**Describe the bugs**  

- Wrong translation on myfaces/api/src/main/resources/javax/faces/Messages_pt.properties and pt_BR for DoubleRangeValidator.NOT_IN_RANGE [[MYFACES-4405](https://issues.apache.org/jira/browse/MYFACES-4405)]
- Possible NPE in CheckDuplicateIdFaceletUtils [[MYFACES-4409](https://issues.apache.org/jira/browse/MYFACES-4409)]
- oam.Flash.REDIRECT should not be set when flash is disabled [[MYFACES-4423](https://issues.apache.org/jira/browse/MYFACES-4423)] 

The following have also been fixed in Liberty are are included as part of this release:
- Implicit 'request' object fails to be resolved by CDI [[MYFACES-4431](https://issues.apache.org/jira/browse/MYFACES-4431)] - https://github.com/OpenLiberty/open-liberty/issues/20762
- Custom NavigationHandler throws NPE during flow handling  [[MYFACES-4432](https://issues.apache.org/jira/browse/MYFACES-4432)] https://github.com/OpenLiberty/open-liberty/issues/21601 
- Memory Leak with ViewScopeBeanHolder's storageMap [[MYFACES-4433](https://issues.apache.org/jira/browse/MYFACES-4433)] https://github.com/OpenLiberty/open-liberty/pull/20951

**Steps to Reproduce**  
Please see release notes. 

**Expected behavior**  
Please see release notes. 

**Diagnostic information:**  
 - OpenLiberty Version: 21.0.0.6 - 23.0.0.5
 - Affected feature(s) `jsf-2.3`
 - Java Version: N/A
 - server.xml configuration N/A

**Additional context**  
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12350093

### PR DESCRIPTION
**Pull request by volosied**
fixes #21501

Last update was on May 4, 2021. 

Overlay Notes: 
 MYFACES-4431 to jsf-2.3 - NavigationHandlerImpl Class
 MYFACES-4433 to 2.3 - ViewScope Classes
 MYFACES-4394 to jsf-2.3 - JsfArtifactProducer & ImplicitObjectResolver
 MYFACES-4450 to 2.2, 2.3, & 3.0 - CommonPropertyUtils

### CODE REVIEW

**Comment by volosied** [comment]
#build

File: dev/com.ibm.websphere.javaee.jsf.2.3/build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2021 IBM Corporation and others.
+ * Copyright (c) 2017, 2022 IBM Corporation and others.
```
Comment (volosied) [inline dev/com.ibm.websphere.javaee.jsf.2.3/build.gradle:2]: Update copyrights to 2023? But these weren't touched in 2023, so they might be ok as is?

File: dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java [hunk 2]
```diff
@@ -75,9 +75,9 @@ public class FATSuite {
         }
     }
 
-    public static final String MOJARRA_API_IMP = "publish/files/mojarra/javax.faces-2.3.9.jar";
-    public static final String MYFACES_API = "publish/files/myfaces/myfaces-api-2.3.9.jar";
-    public static final String MYFACES_IMP = "publish/files/myfaces/myfaces-impl-2.3.9.jar";
+    public static final String MOJARRA_API_IMP = "publish/files/mojarra/javax.faces-2.3.10.jar";
```
Comment (volosied) [inline dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java]: This should be under another issue, as we're only updating MyFaces, not Mojarra.   We can create a separate issue to update the Mojarra jars.

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat

File: dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java (hunk 2 above)
Comment (volosied) [reply dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java]: Should we update? Mojarra's 2.3.10 release is not necessarily aligned to MyFaces'.

File: dev/io.openliberty.faces.internal/bnd.bnd [hunk 3]
```diff
@@ -16,6 +16,8 @@
 
 bVersion=1.0
 
+Implementation-Version: 2.3.10
```
Comment (pnicolucci) [inline dev/io.openliberty.faces.internal/bnd.bnd]: This doesn't look to be used?

**Review by pnicolucci — COMMENTED** [review]
I see you removed some of the source classes in the 2.3 impl project but not all of the o.a.m classes. Can you explain? Are these changes not in 2.3.10?

File: dev/com.ibm.websphere.javaee.jsf.2.3/build.gradle (hunk 1 above)
Comment (pnicolucci) [reply dev/com.ibm.websphere.javaee.jsf.2.3/build.gradle:2]: If the changes for this were pushed up in 2022 then no need to update the copyright.

File: dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd [hunk 4]
```diff
@@ -58,7 +58,7 @@ instrument.classesExcludes: \
   org/**/*.class
 
 Include-Resource: \
-  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.9;EXACT}!/META-INF/**, \
+  @${repo;org.apache.myfaces.core.impl;2.3.10;EXACT}!/META-INF/**, \
```
Comment (pnicolucci) [inline dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd]: Why the change from `org.apache.myfaces.core:myfaces-impl` to `org.apache.myfaces.core.impl`?

File: dev/io.openliberty.faces.internal/bnd.bnd (hunk 3 above)
Comment (volosied) [reply dev/io.openliberty.faces.internal/bnd.bnd]: Ah, interesting. 

This change was from your commit: https://github.com/pnicolucci/open-liberty/commit/a8c0e79cc18096868a70f4523b605144951e6a94#diff-fe0c218a2f00808c842c8190a1f5dd400113f8a5e925ff282c4678c7c81c3d99R17

But this bnd did use `Implementation-Version` previously.  Now we have `com.ibm.ws.org.apache.myfaces.2.3;version=latest,\` instead.

Comment (volosied) [reply dev/io.openliberty.faces.internal/bnd.bnd]: Done under: https://github.com/OpenLiberty/open-liberty/commit/4b6d24b3c22518215bb431376f1668847d113b36

Comment (volosied) [reply dev/io.openliberty.faces.internal/bnd.bnd]: I'll remove this line

File: dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd (hunk 4 above)
Comment (volosied) [reply dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd]: I should revert this as it was a change by Jared: https://github.com/OpenLiberty/open-liberty/commit/a732411f4f4d89464fd4f38a47b317810b29604c#diff-8f1167cf80530cee0c71b8dd6375ad80afe10ecdb3cc197b7b0df74847b33ba9R64

Let me make an update. 
I rebased this branch a while ago, but I don't recall getting conflicts.

**Comment by volosied** [comment]
I removed all overlays except for CommonPropertyUtils -- you can se the diff between the overlay and the 2.3.10 source, which is just [MYFACES-4450](https://github.com/apache/myfaces/pull/300/files#diff-6196fe830f0d8252029e57d9fffd119babf135f6d976cc7f17cbac15da785968R534): 
```
/libertyGit/open-liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/shared/renderkit/html/CommonPropertyUtils.java’
534,538c534
<         if ((commonPropertiesMarked & CommonPropertyConstants.ACCESSKEY_PROP) != 0)
<         {
<             HtmlRendererUtils.renderHTMLStringAttribute(writer, component,
<                     HTML.ACCESSKEY_ATTR, HTML.ACCESSKEY_ATTR);
<         }
—
>         renderAccesskeyTabindexProperties(writer, commonPropertiesMarked, component); 
```

Other overlays were removed as there was not significant code difference: 
```

diff ./org/apache/myfaces/cdi/JsfArtifactProducer.java  ‘/Users/volosied/libertyGit/open-
liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/cdi/JsfArtifactProducer.java’
220,221c220
< 
<     /*
—
>        /*
225d223
< 
227d224
< 
244d240
<    


diff ./org/apache/myfaces/el/unified/resolver/implicitobject/ImplicitObjectResolver.java 
‘/Users/volosied/libertyGit/open-liberty/dev/com.ibm.ws.org.apache.myfaces.2.3/src/org/apache/myfaces/el/unified/
resolver/implicitobject/ImplicitObjectResolver.java’
120c120
< 
—
>  
```

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

File: dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd [hunk 5]
```diff
@@ -58,7 +58,7 @@ instrument.classesExcludes: \
   org/**/*.class
 
 Include-Resource: \
-  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.9;EXACT}!/META-INF/**, \
+  @${repo;org.apache.myfaces.core:myfaces-impl;2.3.10;EXACT}!/META-INF/**, \
```
Comment (volosied) [inline dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd]: Use {Implementation-Version} instead.

File: dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java (hunk 2 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsfContainer_fat_2.3/fat/src/com/ibm/ws/jsf/container/fat/FATSuite.java]: I'm ok with not updating Mojarra here.

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

**Review by pnicolucci — DISMISSED** [review]

**Comment by volosied** [comment]
#libby

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf_fat_lWAS_jsf2.3

**Review by isaacrivriv — APPROVED** [review]

**Comment by volosied** [comment]
#libby
